### PR TITLE
List partial causatives

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

When marking partial causative, variant can be seen on causatives view
Screenshot 2022-03-08 at 10 29 35
and as a matching causative
Screenshot 2022-03-08 at 10 38 44
also, on the variant level:
Screenshot 2022-03-08 at 11 23 27
 
compared to main:
Screenshot 2022-03-08 at 10 41 25
Screenshot 2022-03-08 at 10 41 19
Especially the latter is a little worrying, but afaik at least cust002 rarely use partial.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Review by northwestwitch — APPROVED** [review]
Super useful fix! 🏅 

You might want to check some vulture complaints in mongo/variant.py.

I would make partial causative badge a bit more evident by changing its color and moving it at the beginning of the row. Add also a tooltip to explain what is a partial causative for those who are not already familiar with the concept?

**Comment by northwestwitch** [comment]
What happens in the eventuality that a variant is a causative in a case and a partial causative in another? 🤔

File: scout/adapter/mongo/variant.py [hunk 1]
```diff
@@ -578,7 +577,11 @@ def check_causatives(self, case_obj=None, institute_obj=None, limit_genes=None):
         """
         institute_id = case_obj["owner"] if case_obj else institute_obj["_id"]
         var_causative_events = self.event_collection.find(
-            {"institute": institute_id, "verb": "mark_causative", "category": "variant"}
+            {
+                "institute": institute_id,
+                "verb": {"$in": ["mark_causative", "mark_partial_causative"]},
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/adapter/mongo/variant.py:582]: Something like that; we actually discussed when introducing the feature and postponed it until it saw more use (https://github.com/Clinical-Genomics/scout/pull/1390).

File: scout/server/blueprints/institutes/templates/overview/causatives.html [hunk 2]
```diff
@@ -142,6 +142,9 @@
               </a>
             </td>
             <td>
+              {% if (case.partial_causatives and variant._id in case.partial_causatives) %}
+                <span class="badge badge-secondary">partial</span>
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/institutes/templates/overview/causatives.html:146]: Hm, both the case status and the partial in that case - they belong together. I would say they are good where they are, but explain if Im missing a use case!

I'm no colouring expert so will not press it but my two cents are: 1) its good to keep the number of colors down, much like the number of fonts, but 2) colors should not visually collide with any that is used for quick recognition in the page
3) less important features should have lower signal/contrast value 4) the color should stay roughly the same between pages. Its `badge-secondary` in e.g. the general report. It is not much of a difference when considering a large list, it does not need a high contrast value. 

But, honestly, any of your suggestions is good enough though; I could change them all to e.g. `badge-warning`-yellow instead? Or just keep them for now. 🤷

[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/institutes/templates/overview/causatives.html:146]: I guess I should make a css class instead really. `badge-partial-causative`. 😁

**Comment by dnil** [comment]
> What happens in the eventuality that a variant is a causative in a case and a partial causative in another? 🤔

That should be fine; both (variant, case)-tuples should show in the list. I'll check to make sure!

**Comment by dnil** [comment]
Thanks for the review! I changed the color, added tooltips and fixed Vulture suggestions. 

Screenshot 2022-03-08 at 14 20 10
Screenshot 2022-03-08 at 14 20 38

**Pull request merged**